### PR TITLE
Answer:34

### PR DESCRIPTION
**Pull request by BruceArmstrong007**
## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

Added child components for input and list,
custom ngForEmpty directive to remove ngIf additional block
OnPush Change Detection Added

### CODE REVIEW

File: apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  ngDoCheck(): void {
+    this.ref?.destroy();
+    if (!this.ngForEmptyOf || this.ngForEmptyOf.length === 0) {
+      this.ref = this.vcr.createEmbeddedView(this.ngForEmptyElse);
+    }
+  }
+}
+
+export { NgForEmptyDirective as NgForEmpty };
```
Comment (tomalaforge) [inline apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts:28]: you can now use the new control flow to delete this custom directive 😜

**Comment by tomalaforge** [comment]
Perfect 👍

File: apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts (hunk 1 above)
Comment (BruceArmstrong007) [reply apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts:28]: true, new control flow is better, on V17

Comment (tomalaforge) [reply apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts:28]: you can try it if you want, if you rebase your project and npm ci, you will be on angular v17

**Pull request closed without merging**